app.js

```diff
@@ -166,40 +179,19 @@ api.on('connection', function (spark) {
     }
   });
 
-
-  spark.on('update', function (data) {
-    if (!_.isUndefined(data.id) && !_.isUndefined(data.stats)) {
-      Nodes.update(data.id, data.stats, function (err, stats) {
-        if (err !== null) {
-          console.error('API', 'UPD', 'Update error:', err);
-        } else {
-          if (stats !== null) {
-            client.write({
-              action: 'update',
-              data: stats
-            });
-
-            console.info('API', 'UPD', 'Update from:', data.id, 'for:', stats);
-
-            Nodes.getCharts();
-          }
-        }
-      });
-    } else {
-      console.error('API', 'UPD', 'Update error:', data);
-    }
-  });
-
-
   spark.on('block', function (data) {
     const { stats, proof } = data
-    if (authorize(proof, stats)
+    if (sanitize(stats)
       && !_.isUndefined(stats.block)) {
       stats.id = proof.address
       if (stats.block.validators && stats.block.validators.registered) {
         stats.block.validators.registered.forEach(validator => {
           validator.registered = true
-          trusted.push(validator.address)
+          // trust registered validators and signers - not safe
```

Comment:
could you provide a little more info about why it isn't safe and what we should do to address it in the comment?

Reply:
it's not safe because a reporting node can inject any address that then can inject any address that ...

We need to separate stats (latency, geth events) from state (valset, balances, ...).
Ethstats wants to monitor both, but every node sending smart contract state through the websocket is inefficient, tricky and duplicate work.
This is a longer thread where the goal is to have a trusted and efficient source of state for ethstats. Blockscout graphql is a good candidate.